Fix renderable children

Andrewpk opened the pull request:
This PR loosens the requirements on children and allows any number of elements nested inside a button as long as they're either a string, or valid react elements.

The button `textStyle` property is optional and only applied to bare strings. If you specify child elements that are of `<Text>` type, you'll want to specify your own style on that `<Text>` element. This keeps the original functionality but expands upon it for further customization.

alvaromb commented:
Thanks @Andrewpk!

Will review and test tomorrow. Can you squash your commits??

Andrewpk commented:
@alvaromb sure!

Andrewpk commented:
@alvaromb Also, I think you can have merges default squashed as a repository owner now as well:
https://github.com/blog/2141-squash-your-commits

alvaromb commented:
Sorry for the delay @Andrewpk, too much work happening here right now. Would this close https://github.com/APSL/react-native-button/pull/15, right?

Andrewpk commented:
No worries. I hope all is well. I've been using my PR branch in my project for about a week now. You are correct, it would fix and close #15 as well.

-Andrew

> On May 20, 2016, at 9:54 AM, Álvaro Medina Ballester <notifications@github.com> wrote:
> 
> Sorry for the delay @Andrewpk, too much work happening here right now. Would this close #15, right?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>

Andrewpk commented:
Sorry for wiping out some history here but I'm trying to keep my commits to one as you requested. I recently merged in master.

alvaromb commented:
Hello @Andrewpk!

I've just tested the code, could you please update your PR with the latest changes on master? I can merge today this.

Thanks!

Andrewpk commented:
@alvaromb sorry for the delay - merged in, rebased and pushed!

alvaromb commented:
@Andrewpk the PR looks good, I'm testing it right now.

However the example throws a warning because in the iterator you should define some `key` to the elements:

<img width="487" alt="screen shot 2016-06-27 at 10 35 46" src="https://cloud.githubusercontent.com/assets/289640/16373516/fb1539e4-3c52-11e6-8523-8b9feca04882.png">

Can you update this and check that all tabs are set to two spaces? Squash your commits if you can, too 😊

alvaromb commented:
Just one more thing, can you document these changes into the `README.md`?

Andrewpk commented:
Sure. I forgot I made that change locally and never pushed 😓

Andrewpk commented:
I updated the `README.md` and changed the children requirements a bit. Hopefully my documentation is descriptive enough to be useful.
I changed the looseness of allowing a child to be of type `PropTypes.node` due to potential issues with a javascript array containing numbers or strings being passed as a child. In the current implementation, you can have a `var someNum = 3;` and pass `{someNum}` as a child and it will be properly wrapped in a `<Text>` element, but if you were to pass `var someNums = [3, 4, 5]` as `{someNums}` for a `<Button>`'s child, it would not render them. This could be an area for future enhancement if someone needs this functionality, but for now I felt that the current implementation worked well.

Andrewpk commented:
Had to make one more change, going back to allowing the react type of children to be of type `React.PropTypes.node` otherwise numbers passed to `<Button>` via JSX variables were denied.

alvaromb commented:
Thanks @Andrewpk! Will check and test the code now.

Do you have any cool example that I can insert into the `Example` project and display it in the screenshots?

Andrewpk commented:
@alvaromb The best example I have is using buttons with icons from a vector icon library that uses font awesome. Do you have any issues with me adding an external dependency to the example project in order to bring the icons in?

Andrewpk commented:
@alvaromb as an alternative to pulling in a dependency, I could make a examples with nested `<Image>`s.

alvaromb commented:
Closing by #45.
